### ISSUE TITLE
[Suggestion]: Hosting docs for older Qiskit versions

### ISSUE THREAD

**Issue by drewrisinger** [Qiskit/qiskit-metapackage#933]
## Suggestion for Doc Improvements

Is it possible for Qiskit to host docs for older versions of the Qiskit packages? Downstream packages can't always keep up to date with the ~ monthly release cycle of Qiskit, so it's very useful for end-users to have the older versions documented. [ReadTheDocs supports hosting different versions](https://docs.readthedocs.io/en/stable/versions.html), so I was wondering if the Qiskit team would be able to enable this feature.

Examples of other projects that host multiple documentation versions:
* Scipy
* [Numpy](https://numpy.org/doc/)
* matplotlib

Note: I recognize that end-users can generate documentation themselves, but it's not in as convenient of a format as the meta-documentation provided at https://qiskit.org/documentation/ currently.

## Does existing documentation need to be updated with new content?

No new content needs written. However, this update would likely break currently-bookmarked URLs.

**Comment by mtreinish** [comment]
There is a PR for starting to do this here: https://github.com/Qiskit/qiskit/pull/574 but it's currently blocked by the theme. There isn't a way for the theme to discover the old versions currently. Once that is implemented we can merge that which will build out the old versions of the docs as they were published for each metapackage tag.

**Comment by mtreinish** [comment]
Also as this was implemented in #574 wouldn't break bookmarks because the old versions would be at https://qiskit.org/documentation/stable/0.12/index.html but the current master of Qiskit/qiskit  (which is the docs for the current release of each element and any metapackage doc changes since the last release as opposed to master for everything) would still be https://qiskit.org/documentation/index.html

**Comment by drewrisinger** [comment]
@mtreinish thanks, I didn't check for this in PRs, just issues.

**Comment by nonhermitian** [comment]
It is also a bit more complicated with the tutorials in the docs and the (future) links to the interactive versions in the iqx. That transition now needs to be synced as well.

**Comment by drewrisinger** [comment]
Closing in favor of #574
